Geometry/HGCalCommonData/plugins/dd4hep/DDHGCalHEAlgo.cc

```diff
@@ -0,0 +1,625 @@
+///////////////////////////////////////////////////////////////////////////////
+// File: DDHGCalHEAlgo.cc
+// Description: Geometry factory class for HGCal (Mix)
+// Author : Raman Sehgal
+// DD4hep code for, HGCalHEAlgo, developed by Sunanda Banerjee
+///////////////////////////////////////////////////////////////////////////////
+
+#include <cmath>
+#include <memory>
+#include <string>
+#include <unordered_set>
+#include <vector>
+
+#include "Geometry/HGCalCommonData/interface/HGCalParameters.h"
+#include "Geometry/HGCalCommonData/interface/HGCalGeomTools.h"
+#include "Geometry/HGCalCommonData/interface/HGCalWaferType.h"
+#include "DD4hep/DetFactoryHelper.h"
+#include "DataFormats/Math/interface/CMSUnits.h"
+#include "DetectorDescription/DDCMS/interface/DDPlugins.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+//#define EDM_ML_DEBUG
+using namespace cms_units::operators;
+
+struct HGCalHEAlgo {
```

Comment:
do you need an HGCalHEAlgo object? It looks like all of the constructor code can be copied to the ```algorithm``` function.

Reply:
It could be. But let it be the way it is for now. I shall try to take a look later - but it is more transparent this way

Reply:
Not really. You should throw an exception from a constructor whenever you cannot properly initialize (construct) an object. There is no really satisfactory alternative to exiting a constructor by a throw.